From 3d0390168b5e426129f51fb618cfd60f0a50d1ba Mon Sep 17 00:00:00 2001
From: Martin Jansa <Martin.Jansa@gmail.com>
Date: Sat, 12 Aug 2023 22:41:35 +0200
Subject: [PATCH] CMakeLists.txt: don't install initctl twice and respect
 WEBOS_INSTALL_SBINDIR

* initctl was once installed in /usr/bin/initctl by webos_build_program
  and then again to /sbin/initctl by:
  install(TARGETS  DESTINATION /sbin)

* all utils were installed to /sbin even when sbindir is set to /usr/sbin
  with usrmerge in DISTRO_FEATURES, causing installed-vs-shipped QA:

ERROR: webos-initscripts-1.0-r0 do_package: QA Issue: webos-initscripts: Files/directories were installed but not shipped in any package:
  /sbin/restart
  /sbin/start
  /sbin/status
  /sbin/initctl
  /sbin/stop
Please set FILES such that these items are packaged. Alternatively if they are unneeded, avoid installing them or delete them within do_install.
webos-initscripts: 5 installed and not shipped files. [installed-vs-shipped]

Signed-off-by: Martin Jansa <Martin.Jansa@gmail.com>
---
Upstream-Status: Pending

 src/initctl/CMakeLists.txt | 1 -
 src/utils/CMakeLists.txt   | 5 ++++-
 2 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/src/initctl/CMakeLists.txt b/src/initctl/CMakeLists.txt
index 4803fea..abbb5b8 100644
--- a/src/initctl/CMakeLists.txt
+++ b/src/initctl/CMakeLists.txt
@@ -36,4 +36,3 @@ target_link_libraries(${BIN_NAME} ${LIBS})
 
 # Install
 webos_build_program(NAME ${BIN_NAME})
-install(TARGETS ${BIN_NAME} DESTINATION ${WEBOS_INSTALL_ROOT}/sbin)
diff --git a/src/utils/CMakeLists.txt b/src/utils/CMakeLists.txt
index 3c83d0b..8508a55 100644
--- a/src/utils/CMakeLists.txt
+++ b/src/utils/CMakeLists.txt
@@ -27,4 +27,7 @@ add_executable(status Status.cpp)
 add_executable(restart Restart.cpp)
 
 # Install
-install(TARGETS start stop status restart RUNTIME DESTINATION ${WEBOS_INSTALL_ROOT}/sbin)
+webos_build_program(NAME start ADMIN)
+webos_build_program(NAME stop ADMIN)
+webos_build_program(NAME status ADMIN)
+webos_build_program(NAME restart ADMIN)
